tests/TestCase/ORM/MarshallerTest.php

```diff
@@ -849,6 +849,36 @@ public function testBelongsToManyWithMixedData()
         $this->assertEquals($tagCount + 2, $tags->find()->count());
     }
 
+	/**
+     * Test belongsToMany association with the ForceNewTarget to force saving
+	 * new records on the target tables with BTM relationships when the primaryKey(s)
+	 * of the target table is specified.
+     * 
+     * @return void
+     */
+    public function testBelongsToManyWithForceNew()
+    {
+        $data = [
+            'title' => 'Fourth Article',
+            'body' => 'Fourth Article Body',
+            'author_id' => 1,
+            'tags' => [
+                [
+                    'id'	=> 4,
+                    'name'	=> 'tag4'
+                ]
+            ]
```

Comment:
What would happen if some of the tags existed and others did not? I think that would be useful to capture in the test case.